scripts/dts/gen_defines.py

```diff
@@ -476,29 +476,41 @@ def write_phandle_val_list(dev, entries, ident):
     #
     # entries:
     #   List of entries (two for 'pwms' above). This might be a list of
-    #   edtlib.PWM instances, for example.
+    #   edtlib.PWM instances, for example.  If only one entry is given it
+    #   does not have a suffix '_0', and the '_COUNT' and group initializer
+    #   are not emitted.
     #
     # ident:
     #   Base identifier. For example, "PWM" generates output like this:
     #
     #     #define <device prefix>_PWMS_CONTROLLER_0 "PWM_0"  (name taken from 'label = ...')
     #     #define <device prefix>_PWMS_CHANNEL_0 123         (name taken from #cells in binding)
+    #     #define <device prefix>_PWMS_0 {"PWM_0", 123}
     #     #define <device prefix>_PWMS_CONTROLLER_1 "PWM_1"
     #     #define <device prefix>_PWMS_CHANNEL_1 456
+    #     #define <device prefix>_PWMS_1 {"PWM_1", 456}
+    #     #define <device prefix>_PWMS_COUNT 2
+    #     #define <device prefix>_PWMS {<device_prefix>_PWMS_0, <device_prefix_PWMS_1}
     #     ...
     #
     #   Note: Do not add an "S" to 'ident'. It's added automatically, which
     #   forces consistency.
 
+    init_vals = []
     for i, entry in enumerate(entries):
-        write_phandle_val_list_entry(
-            dev, entry, i if len(entries) > 1 else None, ident)
+        init_vals.append(write_phandle_val_list_entry(
+            dev, entry, i if len(entries) > 1 else None, ident))
+    if len(entries) > 1:
+        out_dev(dev, ident + "S_COUNT", len(init_vals))
+        out_dev(dev, ident + "S", "{" + ", ".join(init_vals) + "}")
 
 
 def write_phandle_val_list_entry(dev, entry, i, ident):
     # write_phandle_val_list() helper. We could get rid of it if it wasn't for
     # write_spi_dev(). Adds 'i' as an index to identifiers unless it's None.
+    # Returns the identifier for the initializer macro.
```

Comment:
A longer explanation or an example might be nice. Was unclear to me what "initializer macro" meant.